h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1646,18 +1743,20 @@ public synchronized boolean compactMoveChunks(int targetFillRate, long moveSize)
         Collections.sort(move, new Comparator<Chunk>() {
             @Override
             public int compare(Chunk o1, Chunk o2) {
-                return Long.signum(o1.block - o2.block);
+                int res = Long.signum(o1.block - o2.block);
+                assert res == Long.compare(o1.block, o2.block);
```

Comment:
This assertion looks strange for me. `Chunk.block` can be negative due to some error? I don't see any other reason for it.

Reply:
leftover, removed